### PR TITLE
Run end to end tests on PRs [SAME VERSION]

### PR DESCRIPTION
**Pull request by bfjelds**
**What this PR does / why we need it**:
Enable end-to-end tests for PRs where containers aren't pushed to ghcr

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

File: .github/workflows/run-test-cases.yml [hunk 1]
```diff
@@ -50,15 +76,15 @@ jobs:
           - Kubernetes
         test-case:
           - end-to-end
-          - helm-install-delete
-          - conservation-of-broker-pods
+          # - helm-install-delete
```
Comment (jiria) [inline .github/workflows/run-test-cases.yml:79]: Curious, why are these getting disabled?

File: .github/workflows/run-test-cases.yml [hunk 2]
```diff
@@ -161,13 +221,14 @@ jobs:
       name: Set sleep duration before running script to 1500
       run: echo 1500 > /tmp/sleep_duration.txt
 
-    # For pull_request and pull_request_target, only main's version.txt
-    # has a Helm chart (the version in the PR does not).  Use origin/main.
+    # For pull_request and pull_request_target, use the local built
```
Comment (jiria) [inline .github/workflows/run-test-cases.yml:224]: nit: locally

File: .github/workflows/run-test-cases.yml [hunk 3]
```diff
@@ -161,13 +221,14 @@ jobs:
...
+    # containers.
     - if: startsWith(github.event_name, 'pull_request')
-      name: Use main version for non-push
+      name: Tell Helm to use the 'local' labels for containers
```
Comment (jiria) [inline .github/workflows/run-test-cases.yml:227]: nit: container images

File: test/shared_test_code.py [hunk 4]
```diff
@@ -31,6 +32,12 @@
 agent_pod_name = ""
 controller_pod_name = ""
 
+def get_extra_helm_args():
+    # Sleep for amount of time specified in SLEEP_DURATION_FILE else don't sleep at all
```
Comment (jiria) [inline test/shared_test_code.py:36]: Is this a TODO comment?

Comment (jiria) [reply test/shared_test_code.py:36]: Or copy & paste by mistake?

Comment (bfjelds) [reply test/shared_test_code.py:36]: copy/paste ... will fix.

File: .github/workflows/run-test-cases.yml (hunk 2 above)
Comment (bfjelds) [reply .github/workflows/run-test-cases.yml:218]: 👍

File: .github/workflows/run-test-cases.yml (hunk 3 above)
Comment (bfjelds) [reply .github/workflows/run-test-cases.yml:221]: 👍

File: .github/workflows/run-test-cases.yml (hunk 1 above)
Comment (bfjelds) [reply .github/workflows/run-test-cases.yml]: the intent was to reduce how much time our workflows are consuming.

* helm-install-delete just installs and deletes helm, the end-to-end already does that, so it is not needed.
* i added the conservation test bits to the end-to-end test, so it is not needed anymore.

by building the containers for the PR tests, we are adding ~15 minutes build time to each test.  not really a big deal for parallel tests in terms of wall-clock time, but might become a problem if we run into the github actions limits,

Comment (bfjelds) [reply .github/workflows/run-test-cases.yml]: they are both deleted now, not disabled.

File: .github/workflows/run-test-cases.yml [hunk 5]
```diff
@@ -84,7 +109,22 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install kubernetes
-    
+
+    - name: Build local containers for PR tests
+      if: startsWith(github.event_name, 'pull_request')
+      env:
+        BUILD_AMD64: 1
+        BUILD_ARM32: 0
```
Comment (kate-goldenring) [inline .github/workflows/run-test-cases.yml:117]: Should/Could we also test on arm?

Comment (bfjelds) [reply .github/workflows/run-test-cases.yml:117]: sounds good ... i don't know how to run a workflow in an arm container.  sounds like a good enhancement :)

File: .github/workflows/run-test-cases.yml [hunk 6]
```diff
@@ -84,7 +109,22 @@ jobs:
...
+        BUILD_ARM64: 0 
+        PREFIX: ghcr.io/deislabs/akri
+        LABEL_PREFIX: pr
+      run: |
+        make akri-build
```
Comment (kate-goldenring) [inline .github/workflows/run-test-cases.yml:122]: Should we delete the images at the end of the test, so we dont have a bunch of pr image in our container registry?

**Review by kate-goldenring — APPROVED** [review]

**Pull request merged**